### PR TITLE
Fixes for download projects and apk generator

### PR DESCRIPTION
**Pull request by bernadettespieler**
Programs with a language version greater than 0.92 could not be downloaded.
APK button was not handled in webview.

### CODE REVIEW

**Comment by avihric** [comment]
ok to test

**Comment by avihric** [comment]
ok to test

**Comment by avihric** [comment]
ok to test

File: catroid/src/org/catrobat/catroid/utils/DownloadUtil.java [hunk 1]
```diff
@@ -124,4 +120,17 @@ protected void onReceiveResult(int resultCode, Bundle resultData) {
 		}
 	}
 
+	public String getProjectNameFromUrl(String url) {
+		int projectNameIndex = url.lastIndexOf(PROJECTNAME_TAG) + PROJECTNAME_TAG.length();
+		String programName = url.substring(projectNameIndex);
+		try {
+			programName = URLDecoder.decode(programName, "UTF-8");
+		} catch (UnsupportedEncodingException e) {
+			Log.e(TAG, "Could not decode program name: " + programName, e);
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/utils/DownloadUtil.java]: If you refactor code, please do not forget about those `e`xception variable shortcuts next time. :ear_of_rice:

File: catroid/src/org/catrobat/catroid/ui/WebViewActivity.java [hunk 2]
```diff
@@ -80,7 +84,21 @@ protected void onCreate(Bundle savedInstanceState) {
 			@Override
 			public void onDownloadStart(String url, String userAgent, String contentDisposition, String mimetype,
 					long contentLength) {
-				DownloadUtil.getInstance().prepareDownloadAndStartIfPossible(WebViewActivity.this, url);
+				Log.e("TAG", "contentDisposition: " + contentDisposition + "   " + mimetype);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/WebViewActivity.java]: `"TAG"` is not really helpful when debugging. It should either be `WebViewActivity.class.getSimpleName()` or a `private static final String TAG = WebViewActivity.class.getSimpleName();` that you use here.

Also it's not an error, so `Log.e` is unwarranted, you might be looking for `.d()` or `i()`.

**Pull request merged**